packages/spatial/src/common/functions/FeathersHooks.tsx

```diff
@@ -357,14 +357,17 @@ export function hashObject(obj) {
  * An internal hook that will listen to realtime updates to a service
  * and update the cache as changes happen.
  */
-export function useRealtime(serviceName: keyof ServiceTypes, refetch: () => void) {
+export function useRealtime(
+  serviceName: keyof ServiceTypes,
+  refetch: (data: any, eventType: 'created' | 'updated' | 'patched' | 'removed') => void
+) {
   useLayoutEffect(() => {
     const service = Engine.instance.api.service(serviceName)
 
-    service.on('created', refetch)
-    service.on('updated', refetch)
-    service.on('patched', refetch)
-    service.on('removed', refetch)
+    service.on('created', (data: any) => refetch(data, 'created'))
```

Comment:
this will not properly clean up the event handlers. they need to be defined as consts and those passed into the off handlers upon unmount of the effect.

Reply:
Fixed!